cmd/processor/app/processor.go

```diff
@@ -174,6 +178,28 @@ func NewProcessor(configurationPath string, platformConfigurationPath string) (*
 
 // Start starts the processor
 func (p *Processor) Start() error {
+
+	// create a goroutine that restarts a trigger if needed
+	go func() {
+		for {
+			select {
+			case triggerInstance := <-p.restartTriggerChan:
+
+				p.logger.DebugWith("Restarting trigger", "triggerInstance", triggerInstance)
+				if err := p.restartTrigger(triggerInstance); err != nil {
+
+					// handle error
+					p.logger.ErrorWith("Failed to restart trigger",
```

Comment:
well, this is a major issue. not being able to restart a trigger means the function is running while trigger is 💀 .
we need to fix it so in case of a restart failure - the processor moves to "failed" state. that means it will reply with "bad" status code to the healthchecks that eventually will lead to a restart.

Reply:
you can achieve that by get trigger instance workers and iterate over each one and "forcely" restart them.

the restart provided here is a "quickier" restart while restarting the trigger's worker is a "hard" one.

P.S - unit test that.

Reply:
Setting the processor's status to `Error` is done by setting the worker's status to `Error`.
However - how can we escalate the restart procedure if a certain worker fails to restart?

Moreover - only an RPC type worker supports restart. What should we do for other types of workers (if they are supported at all)?

Reply:
the last escalation imo is setting each worker in failed state leaving the healthcheck dealing with it.